examples/fluids/index.md

```diff
@@ -351,6 +351,40 @@ $$
 where $(\bar{x}, \, \bar{y}) = (x-x_c, \, y-y_c)$, $(x_c, \, y_c)$ represents the center of the domain, $r^2=\bar{x}^2 + \bar{y}^2$, and $\epsilon$ is the vortex strength ($\epsilon$ < 10).
 There is no perturbation in the entropy $S=P/\rho^\gamma$ ($\delta S=0)$.
 
+(problem-shock-tube)=
+
+## Shock Tube
+
+This test problem is based on Sod's Shock Tube (from{cite}`sodshocktubewiki`), a canonical test case for discontinuity capturing in one dimension. For this problem, the three-dimensional Euler equations are formulated exactly as in the Isentropic Vortex problem. The default initial conditions are $P=1$, $\rho=1$ for the driver section and $P=0.1$, $\rho=0.125$ for the driven section. The initial velocity is zero in both sections. Slip boundary conditions are applied to the side walls and wall boundary conditions are applied at the end walls.
+
+SU upwinding and discontinuity capturing have been implemented into the explicit timestepping operator for this problem. Discontinuity capturing is accomplished using a modified version of the YZB operator described in {cite}`tezduyar2007yzb`. This discontinuity capturing scheme involves the introduction of a dissipation term of the form
+
+$$
+\int_{\Omega} \nu_{SHOCK} \nabla \bm v \!:\! \nabla \bm q dV
+$$
+
+The shock capturing viscosity is implemented following the first formulation described in {cite} `tezduyar2007yzb`. The characteristic velocity $u_{cha}$ is taken to be the acoustic speed while the reference density $\rho_{ref}$ is just the local density. Shock capturing viscosity is defined by the following
+
+$$
+\nu_{SHOCK} = \tau_{SHOCK} u_{cha}^2
+$$
+where,
+$$
+\tau_{SHOCK} = \frac{h_{SHOCK}}{2u_{cha}} \left( \frac{ \,|\, \nabla \rho \,|\, h_{SHOCK}}{\rho_{ref}} \right)^{\beta}
+$$
+
+$\beta$ is a tuning parameter set between 1 (smoother shocks) and 2 (sharper shocks. The parameter $h_{SHOCK}$ is a length scale that is proportional to the element length in the direction of the density gradient unit vector. This density gradient unit vector is defined as $\hat{\bm j} = \frac{\nabla \rho}{|\nabla \rho|}. The original formulation of Tezduyar and Senga relies on the shape function gradient to define the element length scale, but this gradient is not available to qFunctions in libCEED. To avoid this problem, $h_{SHOCK}$ is defined in the current implementation as
```

Comment:
Covariant and contravariant length measures are discussed at the top of the "stabilization scale \tau" drop-down here:
https://libceed.readthedocs.io/en/latest/examples/fluids/#equation-eq-peclet

John Evans says he uses the covariant measure and it's more common among former Hughes students. That's what we use in SU/SUPG.

Reply:
This is measured in the direction of velocity, not the density gradient. Are you suggesting applying the velocity-based measurement or are you suggesting I employ a similar metric based on the density gradient?

Reply:
Yeah, same scale measured in the direction of the density gradient. We should probably make this a helper function (give the Jacobian and a physical-space vector, return covariant length in that direction). Then you could call the same function for SUPG stabilization with velocity and here with the density gradient.

Reply:
Maybe also check if it cancels, since you have the density gradient on the reference element already.

Reply:
If I understand correctly, this would replace the p vector that I have defined in original post for this PR. For the current formulation, this vector is exactly the element length in the x-direction. Would switching to the covariant length scale be more accurate for arbitrary grids? I don't want to use the covariant length scale for h_SHOCK directly because I had to do some tuning of C_YZB to get good behavior for this testcase.

Reply:
If you're on a structured grid, the lengths should be exactly the same. It's when you have non-orthogonal grids that the covariant and contravariant lengths differ.

Reply:
We should make a visual explainer on the website, not specific to the fluids example, because this is a common thing other libCEED users will be thinking about, perhaps for the first time.

Reply:
Helper function is in the code and working as expected. I would appreciate a double check on my indices for `dot(dXdx,vec)` inside the new helper function.